### PR TITLE
Add new op for upgrading Kubelet without draining nodes

### PR DESCRIPTION
**Pull request by yokaze**
Implement this PoC.
https://github.com/cybozu-go/neco/issues/830

Signed-off-by: Daichi Sakaue <daichi-sakaue@cybozu.co.jp>

### CODE REVIEW

File: op/k8s/block_device_move.go [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+		if err != nil {
+			return err
+		}
+
+		agent := inf.Agent(n.Address)
+		if agent == nil {
+			return errors.New("unable to prepare agent for " + n.Nodename())
+		}
+
+		stdout, stderr, err := agent.Run(fmt.Sprintf("find %s -type b", op.CSIBlockDevicePublishDirectory))
```
Comment (ymmt2005) [inline op/k8s/block_device_move.go]: This function would run too slowly when there are many nodes.
Run remote commands in parallel using `well.NewEnvironment` just like other commands do.

File: op/status.go [hunk 2]
```diff
@@ -167,11 +177,76 @@ func GetNodeStatus(ctx context.Context, inf cke.Infrastructure, node *cke.Node,
...
+
+	for _, v := range n.Status.VolumesInUse {
+		// e.g. kubernetes.io/csi/topolvm.cybozu.com^720fab08-e197-4855-ad77-dad24970e3de
+		res := strings.Split(string(v), "^")
+		if len(res) != 2 {
+			continue
+		}
+
+		volumeHandle := res[1]
+		pvList, err := clientset.CoreV1().PersistentVolumes().List(metav1.ListOptions{FieldSelector: "spec.csi.volumeHandle=" + volumeHandle})
```
Comment (dulltz) [inline op/status.go]: It might be now worked

File: op/k8s/block_device_move.go [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+			// 3. Move device file to the new path
+			newDevicePath := filepath.Join(newDirectoryPath, string(po.GetUID()))
+			_, stderr, err = agent.Run(fmt.Sprintf("mv %s %s", tmpDevicePath, newDevicePath))
+			if err != nil {
+				return fmt.Errorf("unable to mv on %s; stderr: %s, err: %v", n.Nodename(), stderr, err)
+			}
+
+			// 4. Fix symlink
+			symlinkSourcePath := filepath.Join(op.CSIBlockDeviceDirectory, pvName, "dev", string(po.GetUID()))
+			_, stderr, err = agent.Run(fmt.Sprintf("ln -nfs %s %s", newDevicePath, symlinkSourcePath))
```
Comment (zoetrope) [inline op/k8s/block_device_move.go]: Isn't it necessary to check the target of symlink?
If `ln` command is failed, the next reconciliation will not execute this `ln` command.

File: op/k8s/block_device_move.go [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+		}
+
+		agent := inf.Agent(n.Address)
+		if agent == nil {
+			return errors.New("unable to prepare agent for " + n.Nodename())
+		}
+
+		stdout, stderr, err := agent.Run(fmt.Sprintf("find %s -type b", op.CSIBlockDevicePublishDirectory))
+		if err != nil {
+			return fmt.Errorf("unable to ls on %s; stderr: %s, err: %v", n.Nodename(), stderr, err)
```
Comment (zoetrope) [inline op/k8s/block_device_move.go]:
```suggestion
			return fmt.Errorf("unable to find on %s; stderr: %s, err: %v", n.Nodename(), stderr, err)
```

File: op/k8s/block_device_link_update.go [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+			if err != nil {
+				return err
+			}
+
+			agent := inf.Agent(n.Address)
+			if agent == nil {
+				return errors.New("unable to prepare agent for " + n.Nodename())
+			}
+
+			stdout, stderr, err := agent.Run(fmt.Sprintf("sudo find %s -maxdepth 1 -name pvc-* -type d", op.CSIBlockDevicePublishDirectory))
```
Comment (ymmt2005) [inline op/k8s/block_device_link_update.go]: CKE does not require the user of SSH can become `root`.
Instead, CKE requires the user to be able to execute `docker run`.

When `root` privilege is needed, CKE should run scripts inside `cke-tools` container.

**Pull request merged**